src/interactive/app/eventloop.rs

```diff
@@ -72,94 +87,221 @@ impl AppState {
+                        Char('O') => self.open_that(tree_view.as_ref()),
+                        Char(' ') => self.mark_entry(
+                            CursorMode::KeepPosition,
+                            MarkEntryMode::Toggle,
+                            window,
+                            tree_view.as_ref(),
+                        ),
+                        Char('d') => self.mark_entry(
+                            CursorMode::Advance,
+                            MarkEntryMode::Toggle,
+                            window,
+                            tree_view.as_ref(),
+                        ),
+                        Char('x') => self.mark_entry(
+                            CursorMode::Advance,
+                            MarkEntryMode::MarkForDeletion,
+                            window,
+                            tree_view.as_ref(),
+                        ),
+                        Char('a') => {
+                            self.mark_all_entries(MarkEntryMode::Toggle, window, tree_view.as_ref())
+                        }
+                        Char('u') | Char('h') | Backspace | Left => {
+                            self.exit_node_with_traversal(tree_view.as_ref())
+                        }
+                        Char('o') | Char('l') | Char('\n') | Right => {
+                            self.enter_node_with_traversal(tree_view.as_ref())
+                        }
+                        Char('H') | Home => self.change_entry_selection(CursorDirection::ToTop),
+                        Char('G') | End => self.change_entry_selection(CursorDirection::ToBottom),
+                        Ctrl('u') | PageUp => self.change_entry_selection(CursorDirection::PageUp),
+                        Char('k') | Up => self.change_entry_selection(CursorDirection::Up),
+                        Char('j') | Down => self.change_entry_selection(CursorDirection::Down),
+                        Ctrl('d') | PageDown => {
+                            self.change_entry_selection(CursorDirection::PageDown)
+                        }
+                        Char('s') => self.cycle_sorting(tree_view.as_ref()),
+                        Char('m') => self.cycle_mtime_sorting(tree_view.as_ref()),
+                        Char('c') => self.cycle_count_sorting(tree_view.as_ref()),
+                        Char('g') => display.byte_vis.cycle(),
+                        _ => {}
+                    },
+                };
+            }
+            self.draw(window, tree_view.as_ref(), *display, terminal)?;
         }
         Ok(ProcessingResult::Finished(WalkResult {
             num_errors: traversal.io_errors,
         }))
     }
+
+    fn tree_view<'a>(&mut self, traversal: &'a mut Traversal) -> Box<dyn TreeView + 'a> {
+        let tree_view: Box<dyn TreeView> = if let Some(glob_source) = &self.glob_mode {
+            Box::new(GlobTreeView {
+                traversal,
+                glob_tree_root: glob_source.tree_root,
```

Comment:
Interesting. Enum may improve few things, e.g. some methods share implementation and I had to extract them outside - this would have been easier with an enum. However, there are some other parts of the code would have been more difficult to do, e.g. all the random places that need the `tree` or the `traversal` to read single property.